code/defines/procs/AStar.dm

```diff
@@ -134,7 +134,10 @@ length to avoid portals or something i guess?? Not that they're counted right no
 
 	open.Enqueue(new /PathNode(start, null, 0, call(start, dist)(end), 0))
 
-	while(!open.IsEmpty() && !path)
+	var/how_many_iter = 0
+	var/sanity = 500 // From some local testing, AStar() tends to hit tick check or succeed around the 450 mark.

```

Comment:
commentary only- very dependent on architecture and load of the host device. I live in fear of the day of byond cpu measures. DUNGMIPS perhaps.